src/storage/src/manifest/storage.rs

```diff
@@ -353,27 +407,69 @@ impl ManifestLogStorage for ManifestObjectStore {
         version: ManifestVersion,
     ) -> Result<Option<(ManifestVersion, Vec<u8>)>> {
         let path = self.checkpoint_file_path(version);
-        match self.object_store.read(&path).await {
-            Ok(checkpoint) => Ok(Some((version, checkpoint))),
-            Err(e) if e.kind() == ErrorKind::NotFound => Ok(None),
-            Err(e) => Err(e).context(ReadObjectSnafu { path }),
-        }
+        // Due to backward compatibility, it is possible that the user's checkpoint not compressed,
+        // so if we don't find file by compressed type. fall back to checkpoint not compressed find again.
+        let checkpoint_data = match self.object_store.read(&path).await {
+            Ok(checkpoint) => {
+                let decompress_data =
+                    self.compress_type
+                        .decode(checkpoint)
+                        .await
+                        .context(DecompressObjectSnafu {
+                            compress_type: self.compress_type,
+                            path,
+                        })?;
+                Ok(Some(decompress_data))
+            }
+            Err(e) if e.kind() == ErrorKind::NotFound => {
+                let uncompress_path = gen_path(
+                    &self.path,
+                    &checkpoint_file(version),
+                    CompressionType::UNCOMPRESSED,
+                );
```

Comment:
issue: The default compression type is uncompressed now. So we won't reach this branch. I think you could add a constant FALL_BACK_COMPRESS_TYPE. Then we just need to switch the value between it and DEFAULT_COMPRESS_TYPE.